cognee/infrastructure/llm/structured_output_framework/baml/baml_src/extraction/create_dynamic_baml_type.py

```
@@ -0,0 +1,122 @@
+        # ------------------------------------------------------------------
+        # 3. Dict / Map -------------------------------------------------------
+        # ------------------------------------------------------------------
+        def _is_enum_subclass(key_type) -> bool:
+            """Guarded issubclass – returns False when tp is not a class."""
+            return isinstance(key_type, type) and issubclass(key_type, Enum)
+
+        if origin in (dict,):
+            key_type, value_type = args or (str, object)
+
+            if key_type is not str and not _is_enum_subclass(key_type):
+                raise ValueError("BAML maps only allow 'str' or Enum subclasses as keys")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)